pkg/cri/sbserver/container_stats_list.go

```diff
@@ -115,6 +115,13 @@ func (c *criService) toCRIContainerStats(
 		if !ok {
 			handler, err = c.getMetricsHandler(ctx, cntr.SandboxID)
 			if err != nil {
+				// If the sandbox is not found, it may have been removed. we need to check container whether it is still exist
+				if errdefs.IsNotFound(err) {
+					_, err = c.containerStore.Get(cntr.ID)
```

Comment:
If err == nil, it should be bug. Trying to reproduce it in my local.

Reply:
After testing, when err == nil, no error will occur and false will be returned. but still added a judgment.